### DISCUSSION TITLE
COMPOSER_HOME/config.json ???

### DISCUSSION THREAD

**Discussion by rino1** [composer/composer#11406] [General]
https://getcomposer.org/doc/03-cli.md#composer-home-config-json
COMPOSER_HOME/config.json
You may put a config.json file into the location which COMPOSER_HOME points to. Composer will partially (only config and repositories keys) merge this configuration with your project's composer.json when you run the install and update commands.
This file allows you to set [repositories](https://getcomposer.org/doc/05-repositories.md) and [configuration](https://getcomposer.org/doc/06-config.md) for the user's projects.
In case global configuration matches local configuration, the local configuration in the project's composer.json always wins.

What does it mean?
However we try

`
$ COMPOSER_HOME/config.json=test
bash: COMPOSER_HOME/config.json=test: No such file or directory
`

???

**Comment by Seldaek** [comment]
you need to check where COMPOSER_HOME is with `composer config -g home`, then inside that directory you can place a `config.json` file.

**Reply by rino1** [reply]
````
$ composer config -g home
[CUT]/.config/composer
````
And now?
You should give me an example, otherwise I don't understand.

**Reply by Seldaek** [reply]
Well i don't quite understand what you're trying to do so maybe explain that first

**Reply by rino1** [reply]
First of all

```
~$ composer config --global home
HOME/.config/composer

~$ \ls .config/composer/config.json 
.config/composer/config.json
```

Logical!!!

https://getcomposer-org.translate.goog/doc/03-cli.md?_x_tr_sl=auto&_x_tr_tl=it&_x_tr_hl=it&_x_tr_pto=wapp#environment-variables

OK!!!
So below are the environment variables including COMPOSER_HOME and COMPOSER_HOME/config.json

In particular…

https://getcomposer-org.translate.goog/doc/03-cli.md?_x_tr_sl=auto&_x_tr_tl=it&_x_tr_hl=it&_x_tr_pto=wapp#composer-home

So COMPOSER_HOME is an environment variable that you can set to your liking.

```
~$ COMPOSER_HOME=/tmp/abc composer config --global home
/tmp/abc
```

Logical!!!

https://getcomposer-org.translate.goog/doc/03-cli.md?_x_tr_sl=auto&_x_tr_tl=it&_x_tr_hl=it&_x_tr_pto=wapp#composer-home-config-json

So COMPOSER_HOME/config.json should also be an environment variable that you can set to your liking.
But…

```
~$ COMPOSER_HOME/config.json=my-config.json composer config --global home
bash: COMPOSER_HOME/config.json=my-config.json: No such file or directory
```

it looks like COMPOSER_HOME/config.jso=my-config.json is being interpreted as a single filename…

I do not understand the use of the COMPOSER_HOME/config.json variable

**Reply by Seldaek** [reply]
You just set COMPOSER_HOME to the directory you want and composer will look for a config.json in there, there is no env var with config.json like you suggest, i don't think env vars can even contain slashes in their name.
